packages/itwinui-css/src/surface/surface.scss

```diff
@@ -22,5 +22,43 @@
 
 // Public facing surface component.
 @mixin iui-surface-component {
+  --iui-surface-padding-inline: var(--iui-size-s);
+  --iui-surface-body-padding: var(--iui-surface-padding-inline);
+
   @include iui-surface;
+  display: flex;
+  flex-direction: column;
+  height: var(--iui-surface-height);
+}
+
+@mixin iui-surface-header-footer($type) {
+  display: flex;
+  flex-shrink: 0;
+  align-items: center;
+  overflow: hidden;
+
+  @if ($type == 'header') {
+    border-bottom: 1px solid var(--iui-surface-border-color);
+    padding: calc(var(--iui-surface-padding-inline) * 0.5) var(--iui-surface-padding-inline);
+  }
+
+  @if ($type == 'footer') {
+    border-top: 1px solid var(--iui-surface-border-color);
+  }
+}
+
+@mixin iui-surface-header {
+  @include iui-surface-header-footer(header);
+}
+
+@mixin iui-surface-footer {
+  @include iui-surface-header-footer(footer);
+}
+
+@mixin iui-surface-body {
+  display: flex;
+  flex-grow: 1;
+  padding: var(--iui-surface-body-padding);
```

Comment:
Because this review is supposed to be from last week, I think it is no longer applicable now. Resolving this.

Reply:
Regarding [the horizontal overflow comment](https://github.com/iTwin/iTwinUI/pull/1044#discussion_r1103251005), I can fix that by adding this:
```diff
@mixin iui-surface-header {
  …

+  > * {
+    overflow: hidden;
+    overflow-wrap: break-word;
+  }
}

@mixin iui-surface-body {
  …
+  overflow-x: hidden;
+
+  > * {
+    overflow: hidden;
+    overflow-wrap: break-word;
+  }

  …
}
```
![Screenshot 2023-02-22 at 4 25 47 PM](https://user-images.githubusercontent.com/849817/220762712-8827830b-b16d-42a8-ae34-f3d6e56d814e.png)

`overflow-x: hidden;` on the body definitely should be added.  It's the `> *` parts that I am unsure about.  Should that fall on the user?  If it doesn't fall on the user, should the styling be applied via surface CSS or via @mayank99's flex component #1011?

Reply:
Regarding [the padding comment](https://github.com/iTwin/iTwinUI/pull/1044#discussion_r1103248549), if I understand what you are asking correctly, you're unsure why the padding in red isn't visually the same as the green or the blue.
![218197710-765c0f28-c689-413c-9b93-4a35841dfa81](https://user-images.githubusercontent.com/849817/220763716-83156130-95a4-4aba-a508-ad94d1dfaa6e.png)

The green side appears to have more padding, but that is only because the text is wrapping unevenly where there's a space in a sentence.  If you right click & inspect the element we can see the padding is equal on both sides despite where the sentence wraps (padding shown in light green).
![Screenshot 2023-02-22 at 4 33 45 PM](https://user-images.githubusercontent.com/849817/220764286-9a775bee-4ec9-494c-8a90-2a32fa39ff40.png) ![Screenshot 2023-02-22 at 4 34 43 PM](https://user-images.githubusercontent.com/849817/220764406-8ef9b568-73b7-4752-8610-7f3f0523ff9a.png)

The blue sides, the padding is set to be half of that on the left & right, so it will be smaller.  The padding is only applied at the top and bottom of the element, even if overflowing is present.  So if you scroll to the center, you may experience text touching the very top / bottom borders.

Reply:
Ah I see. Thanks for explaining!